### DISCUSSION TITLE
DataSet Visualizer Extension

### DISCUSSION THREAD

**Discussion by Sorizz** [microsoft/vscode-discussions#784] [Extension Development QnA] [answered]
Hi Team, 
              I have been working with visual studio since my day 1 of software development. Now i want to switch to vs code for VS Code  is very lightweight and flexible. 
                       But in my work i need to inspect datasets heavily. So in Visual Studio 2019, there's a feature called DataSet Visualizer which helps us to visualise the dataset during debugging in tables.
                       But in vs code I can't find any suitable extensions for that. Can anyone recommend one?

**Comment by theReynald** [comment, marked as answer]
Hello @sorizz,

VS Code doesn't have some features and integrations that full-fledged IDEs like Visual Studio have. The DataSet Visualizer in Visual Studio is one of those features tailored to .NET development that isn't natively replicated in VS Code.

However, if you're transitioning to VS Code and need similar capabilities, consider the following approaches:

1. **.NET Interactive Notebooks**: Microsoft introduced .NET Interactive, which allows you to use .NET in Jupyter notebooks. These notebooks are excellent for data inspection and visualization. You can use the .NET Interactive Notebooks extension in VS Code. While this isn't a direct debugger visualizer, it might be suitable for dataset inspections and manipulations in some cases.

2. **Custom Extension**: Given the extensibility of VS Code, you can always develop (or sponsor the development of) an extension that integrates a dataset visualizer into the VS Code environment. This would be a more involved approach, but it can provide you with precisely what you need.

3. **External Tools**: There are many external tools that allow for dataset visualization, especially if you're working with databases. Tools like DBeaver, DataGrip, and others can connect to databases and provide rich visualization capabilities. You can use these in conjunction with VS Code for your dataset inspection needs.

4. **Feature Requests & Community Engagement**: Engage with the VS Code community on platforms like GitHub. There might be others with similar needs, and voicing your requirements can lead to the development of new extensions or features. Also, checking the Visual Studio Code Marketplace periodically for new extensions is a good idea.

5. **Reconsider the Switch for Specific Tasks**: If the DataSet Visualizer is a crucial part of your workflow and you can't find an equivalent in VS Code, consider using both editors side by side, depending on the task at hand. You can use Visual Studio for tasks that require heavy dataset inspection and VS Code for other tasks.

Hope this helps.
